### PR TITLE
Add Kubeflow Trainer TrainJob example to Run Jobs Using Python docs

### LINKED ISSUE #12634: Introduce use of kubeflow sdk for TrainJob and Kueue integration
@mimowo @kannon92 Any thoughts on introducing Python example w/ Kueue and TrainJob using Kubeflow SDK?
Or you prefer doing this in the followup PR?
Maybe @kaisoz @aniket2405 @abhijeet-dhumal @robert-bell @reckless-sherixx can help with that ?

_Originally posted by @andreyvelich in https://github.com/kubernetes-sigs/kueue/pull/12615#discussion_r3491715428_

### PR DESCRIPTION
**Pull request by kaisoz**
#### What type of PR is this?
/kind documentation
/area website

#### What this PR does / why we need it:
Adds a "Kubeflow Trainer Job" example to the "Run Jobs Using Python" documentation page, showing how to submit a Kubeflow Trainer v2 `TrainJob` managed by Kueue using the Kubeflow SDK.

#### Which issue(s) this PR fixes:

Fixes #12634 

#### Special notes for your reviewer:
**Depends on** #12615 
The example was validated end-to-end on a local kind cluster (Kubeflow Trainer v2 + Kueue with the TrainJob integration enabled). The console output, `kubectl get trainjobs` state, and training logs shown in the doc are from that run.

**AI disclosure:** I used an AI coding assistant (Claude Code) to help draft the example script and documentation. I reviewed all content and validated it on a live cluster before submitting.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

## Summary by CodeRabbit

* **New Features**
  * Added a runnable Python example that submits a Kubeflow Trainer v2 `TrainJob` using the Kubeflow SDK and queue-based scheduling.
  * The example trains a small distributed PyTorch model, streams progress, and prints helpful `kubectl` viewing commands.

* **Documentation**
  * Expanded the “Run Jobs Using Python” guide with a Kubeflow Trainer Job walkthrough, including prerequisites, single-clusterqueue setup, and queue admission via label configuration.

### CODE REVIEW

**Comment by kaisoz** [comment]
@andreyvelich here's a first version, PTAL 😊

File: site/content/en/docs/tasks/run/python_jobs.md [hunk 1]
```diff
@@ -377,3 +377,95 @@ pi is approximately 3.1410376000000002
...
+Now try running the example:
+
+```bash
+python sample-trainjob.py
+```
+```console
+⭐️ Submitting TrainJob to runtime torch-distributed...
+⭐️ Created TrainJob md3b82e943a5
+Use:
+"kubectl get queue" to see queue assignment
```
Comment (amy) [inline site/content/en/docs/tasks/run/python_jobs.md:314]: is queue a shortcut for clusterqueue? Also if you just get it, I dont think it would tell you which one trainjob is assigned to

File: site/content/en/docs/tasks/run/python_jobs.md [hunk 2]
```diff
@@ -377,3 +377,95 @@ pi is approximately 3.1410376000000002
...
+
+For this example, we will use [Kubeflow Trainer](https://www.kubeflow.org/docs/components/trainer/)
+to submit a `TrainJob` (the Kubeflow Trainer v2 API), and specifically the
+[Kubeflow SDK](https://github.com/kubeflow/sdk) to do this easily. Unlike the previous
+examples, the SDK builds and submits the resource for us, so we don't construct the CRD by
+hand. Given our Python environment created in the [setup](#before-you-begin), we can install
+the SDK directly to it as follows:
+
+```bash
+pip install kubeflow
```
Comment (kannon92) [inline site/content/en/docs/tasks/run/python_jobs.md]: https://github.com/kubernetes-sigs/kueue/pull/12615

I was hoping to change this to use uv so its cleaner as a script.

File: site/content/en/docs/tasks/run/python_jobs.md (hunk 1 above)
Comment (kaisoz) [reply site/content/en/docs/tasks/run/python_jobs.md:314]: nope! is a shortcut for `localqueue`. I just followed the convention set by the other examples ([Sample job](https://github.com/kaisoz/kueue/blob/551f82db014905dc2fc778b2c48303248103c8f7/site/content/en/docs/tasks/run/python_jobs.md?plain=1#L81), [Flux Operator Job](https://github.com/kaisoz/kueue/blob/551f82db014905dc2fc778b2c48303248103c8f7/site/content/en/docs/tasks/run/python_jobs.md?plain=1#L191), [MPI Operator Job](https://github.com/kaisoz/kueue/blob/551f82db014905dc2fc778b2c48303248103c8f7/site/content/en/docs/tasks/run/python_jobs.md?plain=1#L325)), but I see how this can be confusing, I can change it if you prefer 😊

File: site/content/en/docs/tasks/run/python_jobs.md (hunk 2 above)
Comment (kaisoz) [reply site/content/en/docs/tasks/run/python_jobs.md]: 🤦🏻 sorry about that! I'm following now the structure you setting in your PR. PTAL!

File: site/content/en/docs/tasks/run/python_jobs.md [hunk 3]
```diff
@@ -377,3 +377,91 @@ pi is approximately 3.1410376000000002
 
 That looks like pi! 🎉️🥧️
 If you are interested in running this same example with YAML outside of Python, see [Run an MPIJob](/docs/tasks/run/kubeflow/mpijobs/).
+
+### Kubeflow Trainer Job
+
+For this example, we will use [Kubeflow Trainer](https://www.kubeflow.org/docs/components/trainer/)
```
Comment (andreyvelich) [inline site/content/en/docs/tasks/run/python_jobs.md]:
```suggestion
For this example, we will use [Kubeflow Trainer](https://trainer.kubeflow.org/en/latest/)
```

File: site/content/en/docs/tasks/run/python_jobs.md [hunk 4]
```diff
@@ -377,3 +377,91 @@ pi is approximately 3.1410376000000002
...
+to submit a `TrainJob` (the Kubeflow Trainer v2 API), and specifically the
+[Kubeflow SDK](https://github.com/kubeflow/sdk) to do this easily. Unlike the previous
+examples, the SDK builds and submits the resource for us, so we don't construct the CRD by
+hand. The `kubeflow` dependency is declared in the script's inline metadata and will be
+automatically installed by `uv run`.
+
+This example assumes Kubeflow Trainer v2 is installed, the built-in `torch-distributed`
+`ClusterTrainingRuntime` is available, and Kueue is configured to manage TrainJobs. For these
+prerequisites (installing Kubeflow Trainer, the runtime, and enabling the TrainJob
+integration in Kueue), follow the [Run A TrainJob](/docs/tasks/run/trainjobs/) guide. You'll
```
Comment (andreyvelich) [inline site/content/en/docs/tasks/run/python_jobs.md:290]: It would be nice to update the Kueue guide with this new Trainer website in the followup PR: https://trainer.kubeflow.org/en/latest/
cc @Sridhar1030  @h0pers

File: site/static/examples/python/sample-trainjob.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    """
+    parser = get_parser()
+    args, _ = parser.parse_known_args()
+
+    client = TrainerClient()
+    runtime = client.get_runtime(name=args.runtime)
+
+    print(f"⭐️ Submitting TrainJob to runtime {args.runtime}...")
+    job_name = client.train(
+        runtime=runtime,
```
Comment (andreyvelich) [inline site/static/examples/python/sample-trainjob.py]: You can directly set it as:

```suggestion

    print(f"⭐️ Submitting TrainJob to runtime {args.runtime}...")
    job_name = client.train(
        runtime=args.runtime,
```

File: site/content/en/docs/tasks/run/python_jobs.md [hunk 6]
```diff
@@ -377,3 +377,91 @@ pi is approximately 3.1410376000000002
...
+training finishes.
+
+A TrainJob runs on top of a JobSet, so its training pods carry the
+`jobset.sigs.k8s.io/jobset-name` label set to the TrainJob name. You can follow the
+training output:
+
+```bash
+kubectl logs -l jobset.sigs.k8s.io/jobset-name=md3b82e943a5 --tail=-1
+```
+```console
```
Comment (andreyvelich) [inline site/content/en/docs/tasks/run/python_jobs.md]: Instead of using `kubectl` get we use Kubeflow SDK APIs to get, watch, and print logs of a TrainJob?
You can check example in this Notebook:
https://github.com/kubeflow/trainer/blob/master/examples/pytorch/image-classification/mnist.ipynb

**Review by andreyvelich — COMMENTED** [review]
Thanks for this @kaisoz!
I left a few comments.

**Review by andreyvelich — COMMENTED** [review]
Thanks @kaisoz!
/lgtm

**Comment by kaisoz** [comment]
@andreyvelich Thanks! I had to rebase `main`.. could you re-`lgtm` please? 👼🏻

**Comment by mimowo** [comment]
/approve
/cherrypick website
Thank you for the example 👍 I don't have any useful comments, let's iterate on the doc once it is live if folks have ideas for improvements
